test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+        defaults[0] = bob;
+        forwarderFactory = new ERC20ForwarderFactory();
+        factory = new HyperdriveFactory(
+            HyperdriveFactory.FactoryConfig({
+                governance: alice,
+                hyperdriveGovernance: bob,
+                feeCollector: celine,
+                sweepCollector: sweepCollector,
+                defaultPausers: defaults,
+                checkpointDurationResolution: 1 hours,
+                minCheckpointDuration: 8 hours,
+                maxCheckpointDuration: 1 days,
+                minPositionDuration: 7 days,
+                maxPositionDuration: 10 * 365 days,
+                minFixedAPR: 0.001e18,
+                maxFixedAPR: 0.5e18,
+                minTimeStretchAPR: 0.005e18,
+                maxTimeStretchAPR: 0.5e18,
+                minFees: IHyperdrive.Fees({
+                    curve: 0,
+                    flat: 0,
+                    governanceLP: 0,
+                    governanceZombie: 0
+                }),
+                maxFees: IHyperdrive.Fees({
+                    curve: ONE,
+                    flat: ONE,
+                    governanceLP: ONE,
+                    governanceZombie: ONE
+                }),
+                linkerFactory: address(forwarderFactory),
+                linkerCodeHash: forwarderFactory.ERC20LINK_HASH()
+            })
+        );
+
+        poolConfig = IHyperdrive.PoolDeployConfig({
+            baseToken: config.baseToken,
+            linkerFactory: factory.linkerFactory(),
+            linkerCodeHash: factory.linkerCodeHash(),
+            minimumShareReserves: 1e15,
+            minimumTransactionAmount: config.minTransactionAmount,
+            positionDuration: config.positionDuration,
+            checkpointDuration: CHECKPOINT_DURATION,
+            timeStretch: 0,
+            governance: factory.hyperdriveGovernance(),
+            feeCollector: factory.feeCollector(),
+            sweepCollector: factory.sweepCollector(),
+            fees: IHyperdrive.Fees({
+                curve: 0,
+                flat: 0,
+                governanceLP: 0,
+                governanceZombie: 0
+            })
+        });
+    }
+
+    /// @dev A virtual function that exposes important state
+    ///      about the underlying protocol and token. Not all values
+    ///      will be used depending on the instance.
+    /// @param baseSupply Amount of base tokens in circulation.
+    /// @param shareSupply Amount of share tokens in circulation.
+    /// @param sharePrice Price of of one share in terms of base.
+    function getProtocolSharePrice()
+        internal
+        virtual
+        returns (uint256 baseSupply, uint256 shareSupply, uint256 sharePrice);
+
+    /// @dev A virtual function that defines the deployer coordinator
+    ///      contract that will be used to deploy all the instance targets.
+    function deployCoordinator() internal virtual returns (address);
```

Comment:
Right now, these functions are sort of buried between non-abstract functions in the test. It would be better IMO to create a section of the functions that need to be overrided after setup. It would be good to separate these sections with comments like:

```
/// Overrides ///
```

and 

```
/// Tests ///
```

or something like that.